cli/cmd/stage/stage.go

```diff
@@ -85,6 +89,11 @@ func (r *Stage) Run(ctx *clicontext.CLIContext) error {
 		if ctx.CLI.String("image") != "" {
 			spec.Image = ctx.CLI.String("image")
 		}
+		stageEnvs, err := stringers.ParseAllEnv(r.EnvFile, r.Env, true)
+		if err != nil {
+			return err
+		}
+		spec.Env = append(spec.Env, stageEnvs...)
```

Comment:
Looks like this will only add new envs but not replacing existing envs

Reply:
Hmm, we do need to replace. I'm guessing we want to carry all old ones forward and merge with new ones, but we also want to overwrite any old ones.